integration_tests/src/main/python/collection_ops_test.py

```diff
@@ -17,10 +17,48 @@
 from asserts import assert_gpu_and_cpu_are_equal_collect
 from data_gen import *
 from pyspark.sql.types import *
+from string_test import mk_str_gen
+import pyspark.sql.functions as f
 
 nested_gens = [ArrayGen(LongGen()),
                StructGen([("a", LongGen())]),
                MapGen(StringGen(pattern='key_[0-9]', nullable=False), StringGen())]
+# additional test for NonNull Array because of https://github.com/rapidsai/cudf/pull/8181
+non_nested_array_gens = [ArrayGen(sub_gen, nullable=nullable)
+                         for nullable in [True, False] for sub_gen in all_gen + [null_gen]]
+
+@pytest.mark.parametrize('data_gen', non_nested_array_gens, ids=idfn)
+def test_concat_list(data_gen):
+    assert_gpu_and_cpu_are_equal_collect(
+        lambda spark: binary_op_df(spark, data_gen).selectExpr('concat(a)'))
+
+    assert_gpu_and_cpu_are_equal_collect(
+        lambda spark: binary_op_df(spark, data_gen).selectExpr('concat(a, b)'))
+
+    assert_gpu_and_cpu_are_equal_collect(
+        lambda spark: three_col_df(spark, data_gen, data_gen, data_gen
```

Comment:
Sorry I forgot this before. We need to test with literal arrays as well. with concat. There was nothing in the checks that would have prevented concat from supporting literal arrays so we need to test them.

Reply:
For now, concat on `GpuScalar` is also supported. And corresponding tests are appended.